eel/src/lib.rs

```diff
@@ -402,34 +402,55 @@ impl LightningNode {
     }
 
     pub fn pay_invoice(&self, invoice: String) -> Result<()> {
-        let invoice = Self::parse_validate_invoice(self, &invoice)?;
+        let invoice_struct = Self::parse_validate_invoice(self, &invoice)?;
 
-        let amount_msat = invoice
+        let amount_msat = invoice_struct
 	    .amount_milli_satoshis()
 	    .ok_or_invalid_input("Invalid invoice - invoice is a zero value invoice and paying such invoice is not supported yet")?;
 
-        match self.invoice_payer.pay_invoice(&invoice) {
+        let mut payment_store = self.payment_store.lock().unwrap();
```

Comment:
It shouldn't take much time as this doesn't actually pay the invoice but only starts the process. Even so, as we don't have a need to keep it locked, I have refactored in [f603df3](https://github.com/getlipa/lipa-lightning-lib/pull/217/commits/f603df3ae2c58cedb5d7d4b7b791f6a80e0f1037) so that the lock isn't kept for so long.